integration/walletextension/wallet_extension_test.go

```diff
@@ -456,36 +501,42 @@ func makeHTTPEthJSONReq(method string, params interface{}) []byte {
 }
 
 // Makes an Ethereum JSON RPC request over websockets and returns the response body as JSON.
-func makeWSEthJSONReqAsJSON(method string, params interface{}) map[string]interface{} {
-	respBody := makeWSEthJSONReq(method, params)
-	return convertRespBodyToJSON(respBody)
+func makeWSEthJSONReqAsJSON(method string, params interface{}) (map[string]interface{}, *websocket.Conn) {
+	respBody, conn := makeWSEthJSONReq(method, params)
+	return convertRespBodyToJSON(respBody), conn
 }
 
 // Makes an Ethereum JSON RPC request over websockets and returns the response body.
-func makeWSEthJSONReq(method string, params interface{}) []byte {
+func makeWSEthJSONReq(method string, params interface{}) ([]byte, *websocket.Conn) {
 	conn, resp, err := websocket.DefaultDialer.Dial(walletExtensionAddrWS, nil)
 	if resp != nil && resp.Body != nil {
 		defer resp.Body.Close()
 	}
-	if conn != nil {
-		defer conn.Close()
```

Comment:
Annoyingly, we can no longer just `defer` closing the connection, as we need it to stay open if nothing goes wrong so that we can listen for logs.